### DISCUSSION TITLE
Change favicon based on color theme

### DISCUSSION THREAD

**Discussion by aar0npal** [vercel/next.js#54433] [Help] [answered]
I want to change the icon of the tab depending on the color scheme (light or dark), is that possible in App Router like the following?
```tsx
export const metadata: Metadata = {
  title: 'Site',
  icons: {
    icon: isDark ? '/favicon-dark.svg' : 'favicon.svg',
  },
}
```

**Comment by devjiwonchoi** [comment]
Could you share how you are detecting the current scheme in your app?

**Reply by aar0npal** [reply]
I don't know how to detect it yet. `isDark` is just a placeholder or a conceptual representation of a potential feature

**Reply by aar0npal** [reply, marked as answer]
I finally got it thanks to @eric-burel in our discussion in the Discord server

```tsx
export const metadata: Metadata = {
  title: 'Site',
  icons: [
    {
      rel: 'icon',
      type: 'image/x-icon',
      url: '/favicon.ico',
      media: '(prefers-color-scheme: light)',
    },
    {
      rel: 'icon',
      type: 'image/png',
      url: '/favicon-dark.png',
      media: '(prefers-color-scheme: dark)',
    },
  ],
}
```

**Reply by devjiwonchoi** [reply]
Thanks for sharing! 
A bit of improvement could be as follows:

```ts
icons: {
  icon: [
    {
      url: '/light-icon.png',
      media: '(prefers-color-scheme: light)',
      // ...rest
    },
    {
      url: '/dark-icon.png',
      media: '(prefers-color-scheme: dark)',
      // ...rest
    },
  ],
},
```

Note that in this case, you might have to remove `favicon.ico` from your `app/`.

**Reply by aar0npal** [reply]
I see. Thank you!

**Comment by miketdonahue** [comment]
Can anyone confirm this is still working? Doesn't seem to be on the latest Next.js.

To be specific, the "dark" icon doesn't seem to show in Mac System "Dark" mode.

**Reply by kristiankauffeld** [reply]
I have the same issue, but for me it's the "light" icon that doesn't show when in Mac System "Light" mode...

**Comment by rakibulinux** [comment]
> Can anyone confirm this is still working? Doesn't seem to be on the latest Next.js.
> 
> To be specific, the "dark" icon doesn't seem to show in Mac System "Dark" mode.
This one is working for me.


```
<div className="flex justify-center text-teal-600 sm:justify-start">
              <Image
                className="hidden dark:block"
                src={carRapair}
                alt="car repair service"
                width={200}
                height={600}
              />
              <Image
                className="mb-4 block dark:hidden"
                src={carRapairBlack}
                alt="car repair service"
                width={200}
                height={600}
              />
            </div>
```

**Comment by kristiankauffeld** [comment]
I just tested the solution for this thread in 3 browsers (Chrome, Firefox, and Brave), and it only works in Brave. Does anyone have a clue why?

**Reply by kristiankauffeld** [reply]
Just had to restart Chrome then it worked...

**Reply by rakibulinux** [reply]
Awesome bro

**Comment by felipepalazzo** [comment]
I'm not using the app directory yet. Does anyone know how to make this work using page router?

**Reply by felipepalazzo** [reply]
For those still using page router, this worked for me: 

```js
      <link
        href="/favicon.ico"
        rel="shortcut icon"
        type="image/x-icon"
        media="(prefers-color-scheme: light)"
      />
      <link
        href="/static/images/icon.png"
        rel="icon"
        type="image/png"
        sizes="32x32"
        media="(prefers-color-scheme: dark)"
      />
```

**Comment by DanilKasianenko** [comment]
Hello, 

That my way how I do this in `next: 14.0.2` 

```tsx
import { Metadata } from "next";

import DarkIcon from "@/images/dark/logo-dark.png";
import LightIcon from "@/images/light/logo-light.png";

export const metadata: Metadata = {
  title: "Your Title",
  icons: {
    icon: [
      {
        rel: "icon",
        media: "(prefers-color-scheme: light)",
        type: "image/png",
        url: LightIcon.src,
      },
      {
        rel: "icon",
        media: "(prefers-color-scheme: dark)",
        type: "image/png",
        url: DarkIcon.src,
      },
    ],
  },
};
```

NextJs Optimizing Static Assets url to files in public folder, and for that I take current src on files that I need

**Comment by barziahmed** [comment]
I have played with SVG styling with icon.svg, [similar to this](https://web.dev/articles/building/an-adaptive-favicon), but it didn't work. Why doesn't inline styling work in icon.svg?

* I use latest version of Next.js with App router.

**Reply by BenedictLang** [reply]
I would love to know that as well.

I removed the favicon and placed the svg inside. It doesnt seem to load when I launch the site, but after navigating it does...

**Comment by musjj** [comment]
On Firefox, toggling the dark/light simulation in the web dev tools doesn't change the icon colors. I tested the same method on GitHub and it works. Also tested it with Next.js + Chrome and it also works. So, I'm guessing there's an issue specific to Next.js here.

EDIT: Looks like that this might be a Firefox bug: https://stackoverflow.com/questions/71883672/firefox-not-respecting-user-mediaprefers-color-scheme-dark
There's an old fixed bug report: https://bugzilla.mozilla.org/show_bug.cgi?id=1772632, but I think there's a regression. Also, GitHub seems to be setting the icon color with another method, probably with JS.
